src/components/code-editor/code-editor.tsx

```diff
@@ -0,0 +1,44 @@
+import { useCallback, useState } from 'react';
+import Editor, { Theme, EditorProps } from '@monaco-editor/react';
+
+interface CodeEditorProps {
+    code?: string,
+    onChange?: (code: string) => void,
+    language?: string,
+    theme?: Theme,
+    height?: string,
+    options?: EditorProps['options'],
+}
+
+const CodeEditor: React.FC<CodeEditorProps> = (props) => {
```

Comment:
I think it would better to destruct the props and set the default values instead of scattering it all over the code
```suggestion
const CodeEditor: React.FC<CodeEditorProps> = ({
height = '90vh',
theme = 'vs-dark'
}) => {
```

Reply:
Updated, please take a look!